cfme/tests/cloud/test_providers.py

```diff
@@ -1256,7 +1256,19 @@ def test_ec2_create_sns_topic():
             1.
             2. Check SNS topic for this region in AWS
     """
-    pass
+    # preparations for test
+    request.addfinalizer(lambda: provider.delete_if_exists())
+    topic = provider.mgmt.get_arn_if_topic_exists('AWSConfig_topic')
```

Comment:
I am confused
```
             2. Check SNS topic for this region in AWS
```
If the point of this test is checking the topic should exists, you need to assert it is present and not just delete it immediately.

If you need clean environment before the test does the actual testing, then I think either you need some janitor job somewhere which would clean-up old stuff or I think less-preferably you need to put the code deleting the unwanted elements in the fixture.